src/FacesRatingBar/FacesRatingBar.js

```diff
@@ -0,0 +1,275 @@
+import FaceGrining from 'wix-ui-icons-common/dist/src/general/dist/components/FaceGrining';
+
+import Box from '../Box/Box';
+import Tooltip from '../Tooltip/Tooltip';
+
+const faceIconsMap = {
+  1: FaceDisapointed,
+  2: FaceFrowning,
+  3: FaceNeutral,
+  4: FaceSmiling,
+  5: FaceGrining,
+};
+
+/** A rating component that will enable the user to rate on a 1-5 scale. */
+class FacesRatingBar extends React.PureComponent {
+  state = {
+    faceHoveredIndex: 0,
+  };
+
+  componentDidMount() {
+    const { readOnly, value } = this.props;
+
+    if (readOnly && value === 0) {
+      throw new Error(
+        'In readOnly mode the value couldn’t be 0. Please enter a value between 1 to 5.',
+      );
+    }
+  }
+
+  _onFaceClick = index => {
+    this.props.onChange(index);
+  };
+
+  _onFaceMouseEnter = index => {
+    this.setState({ faceHoveredIndex: index });
+  };
+
+  _onFaceMouseLeave = () => {
+    this.setState({ faceHoveredIndex: 0 });
+  };
+
+  _onFaceFocus = (faceIndex, focusableProps) => {
+    // We would like to change the faces color on focus / hover
+    this.setState({ faceHoveredIndex: faceIndex });
+
+    focusableProps.focusableOnFocus();
+  };
+
+  _onFaceBlur = focusableProps => {
+    // We would like to change the faces color on focus / hover
+    this.setState({ faceHoveredIndex: 0 });
+
+    focusableProps.focusableOnBlur();
+  };
+
+  _shouldShowDescriptionValues = () => {
+    const { readOnly, descriptionValues } = this.props;
+    let shouldShowDescriptionValues = false;
+
+    // Adding description values is not available in read only mode and it must be an array of strings at size 5.
+    if (!readOnly && descriptionValues) {
+      shouldShowDescriptionValues =
+        Array.isArray(descriptionValues) && descriptionValues.length === 5;
+    }
+
+    return shouldShowDescriptionValues;
+  };
+
+  render() {
+    const { dataHook, readOnly, size, value, descriptionValues } = this.props;
+    const { faceHoveredIndex } = this.state;
+
+    const showDescriptionValues = this._shouldShowDescriptionValues();
+
+    return (
+      <Box {...styles('root', {}, this.props)} data-hook={dataHook}>
+        <Faces
+          readOnly={readOnly}
+          size={size}
+          selectedIndex={value}
+          hoveredIndex={faceHoveredIndex}
+          showDescriptionValues={showDescriptionValues}
+          descriptionValues={descriptionValues}
+          onClick={this._onFaceClick}
+          onMouseEnter={this._onFaceMouseEnter}
+          onMouseLeave={this._onFaceMouseLeave}
+          handleFocus={this._onFaceFocus}
+          handleBlur={this._onFaceBlur}
+        />
+      </Box>
+    );
+  }
+}
+
+const Faces = ({
+  readOnly,
+  size,
+  selectedIndex,
+  hoveredIndex,
+  showDescriptionValues,
+  descriptionValues,
+  onClick,
+  onMouseEnter,
+  onMouseLeave,
+  handleFocus,
+  handleBlur,
+}) => {
+  return Object.values(faceIndexes).map(faceIndex => {
```

Comment:
Optional

```suggestion
  return Array(5).fill(0).map((_, index)=> {
```
and then use the index
or use a common for loop

Reply:
I changed it but I think it was more readable before these changes